### PR TITLE
Update `test_recurring_job` story

### PR DESCRIPTION
**Pull request by roger-ryao**
longhorn/longhorn#9453

#### Which issue(s) this PR fixes:
longhorn/longhorn#9453

#### What this PR does / why we need it:
- Updated the number of backups completed for `backup1` and `backup2`.
- Adjusted the timeline for data writes and backup intervals.
- Changed the sleep time from 20 to 30 seconds to align snapshot and backup timing.
- Clarified the expected number of backups and added a comment about the volume-head snapshot.

#### Additional documentation or context
https://ci.longhorn.io/job/public/job/v1.6.x/job/v1.6.x-longhorn-upgrade-tests-sles-amd64/196/testReport/junit/tests/test_recurring_job/test_recurring_job_nfs_/

### CODE REVIEW

File: manager/integration/tests/test_recurring_job.py [hunk 1]
```diff
@@ -241,15 +241,16 @@ def test_recurring_job(set_random_backupstore, client, volume_name):  # NOQA
                                          groups=[DEFAULT])
 
     wait_until_begin_of_an_even_minute()
-    # wait until the 20th second of an even minute
+    # wait until the 30th second of an even minute
     # make sure that snapshot job happens before the backup job
-    time.sleep(20)
+    time.sleep(30)
```
Comment (yangchiu) [inline manager/integration/tests/test_recurring_job.py:246]: Will adjusting this sleep time affect the pass rate of this test case?

Comment (roger-ryao) [reply manager/integration/tests/test_recurring_job.py:246]: Yes, I have executed the private image test 10 times, and the result passed on v1.6.3-rc1:
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7587/testReport/tests/test_recurring_job/. 
Screenshot_20240912_155328


However, I must admit that, at the moment, I don't have any clear ideas about what increasing the time would actually improve.

Comment (yangchiu) [reply manager/integration/tests/test_recurring_job.py:246]: That’s fine. It’s possible that the process occasionally takes longer than 20 seconds to initiate.

**Review by yangchiu — APPROVED** [review]

**Comment by roger-ryao** [comment]
@mergify backport v1.6.x v1.7.x

**Pull request merged**